impl/src/planning.rs

```diff
@@ -136,7 +140,7 @@ impl CrateCatalogEntry {
     is_workspace_crate: bool,
   ) -> Self {
     let sanitized_name = package.name.replace("-", "_");
-    let sanitized_version = util::sanitize_ident(&package.version);
+    let sanitized_version = package.version.clone();
```

Comment:
I think this is wrong, but your tests are passing so I'm really confused.

util::sanitize_ident was not just concerned with type conversions. It also [removed funny punctuation and converted dashes to underscores](https://github.com/google/cargo-raze/blob/8337c7b5113aad3de2038bc89ef23d47b82c35e6/impl/src/util.rs#L164-L166). I guess it's possible that we lack tests that cover this behavior.

Do you know if this is being done for us by cargo_metadata?

Reply:
cargo_metadata uses a semvar type for Version which I believe uses type-safety to prevent invalid punctuation: 
https://docs.rs/cargo_metadata/0.9.1/cargo_metadata/struct.Package.html
https://docs.rs/semver/0.9.0/semver/struct.Version.html

Reply:
Following-up here: I think you're 100% right! The smoke tests appear to be failing with this patch, and the difference is "dot" vs "underscore". I'll add a commit that fixes this.

Reply:
This has been fixed, and test coverage has been added checking that the workspace path does not contain any "." or "-" characters.